### PR TITLE
Listening Post 2.0

### PR DESCRIPTION
**Pull request by DangerRevolution**
## About the PR

- Adds Syndicate pAI to keep the crew company. 
- Adds a Synthesiser as people asked for an instrument aboard the vessel
- Adds a second voicemask to the station
- Adds new Syndicate Fax Machine, that is unable to fax to Central Command (pending merge by VMSolidus)
- Adds one-way handheld radio; where the crew aboard can receive messages (I'm open to remove this)
- Lowers volume of the Syndicate Crew Monitoring Server as it is insanely deafening to be around with no other noise going on.
- Listening Post now spawns in the same `instance` as the station, surrounded by debris.

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

image
image

printing
icon
idle
inserting
image

:cl: DangerRevolution and VMSolidus
- Add: Adds pAI, one-way handheld radio, instrument and new fax machine to the Listening Post
- Add: All Syndicate Operatives now spawn with a Cybersun Pen
- Remove: Syndicate Listening Post Operative no longer requires whitelist
- Add: Listening Post now spawns visible on IFF.
- Add: The Listening Outpost has received significant improvements to its interiors. It's now a place I personally would feel comfortable working in-VMSolidus

### CODE REVIEW

**Comment by DebugOk** [comment]
I want to review the map changes before this gets merged, just an fyi to Colin

**Comment by Colin-Tel** [comment]
This must be the listening post post-hotfix fix :trollface:

**Comment by DangerRevolution** [comment]
> This must be the listening post post-hotfix fix :trollface:

honestly this was something meant to be included earlier; I was just battling Visual Studio to stop starting maps I was saving

**Comment by DangerRevolution** [comment]
Colour of fax machine is temporary, gonna switch it back to the regular colour for now to keep up the "stolen from nanotrasen" theme.

**Comment by VMSolidus** [comment]
I've added my new spawning mechanic for the Listening Outpost. 
TODO: Also spawn in random debris, and change the name that the outpost appears as on the mass scanner.

**Comment by VMSolidus** [comment]
https://github.com/DeltaV-Station/Delta-v/assets/16548818/3a47d719-1c81-41f2-a3c3-a1714a879a5d

In addition to some more map tweaks(and tweaks to the outpost self destruct system), I've also implemented some lighting improvements for colorblind players, by request from an actual colorblind player. The outpost now has both red and white lighting, as well as buttons to selectively toggle either light color.

**Comment by VMSolidus** [comment]
- The Syndicate has taken note of crew hostile actions towards the outpost, and has changed the type of asteroids they use to make the outposts. 
- "Mindwipes" will no longer permit their combat equipment to be confiscated while they're alive
- "Mindwipes" will now trigger their death acidifiers immediately on crit to prevent capture of equipment
- TODO: Actually spawn the debris field

**Comment by VMSolidus** [comment]
- The listening outpost no longer has a minimum time for it to spawn
- Outpost operators no longer require whitelist.
Since I've patched out the underlying reasons why we had these two, they're no longer necessary. So now we can open up the role to whitelisted players. We needed whitelist for both testing purposes, and to reduce the chances of players trying to take the guards to the station. We should probably also consider reducing the job time requirements for them, especially for Security, since these are largely RP roles anyway. 

As for the minimum time for the event to trigger, we originally put that in because we were worried about Listening Posts FTL'ing to Nukieworld. Which was still possible when they had a shuttle console. We couldn't prevent them from doing so at the time, so instead as a stopgap until I could code the new spawning system, we made it so that it could only spawn in roughly around the time Nukies normally enter the station.

**Comment by DebugOk** [comment]
> we made it so that it could only spawn in roughly around the time Nukies normally enter the station.

Why not just remove the whitelist on the console? Having a seperate prototype is required anyways for when the FTL distance inevitably gets changed. And I like having a minimum time, it shouldn't happen if the first 15-30 minutes of the round starting

**Comment by VMSolidus** [comment]
> Why not just remove the whitelist on the console? Having a seperate prototype is required anyways for when the FTL distance inevitably gets changed. And I like having a minimum time, it shouldn't happen if the first 15-30 minutes of the round starting

We don't have the shuttle console at all anymore, and I've already implemented the new spawning system. It no longer spawns in nullspace at all, and instead spawns as we originally intended, roughly ~1 to ~1.5km away from the station. The shuttle console was replaced with a mass scanner, since the station doesn't need to FTL anymore. Right now I'm hard at work coding the spawning system for its debris field.

**Comment by VMSolidus** [comment]
I'll add back in a modest minimum timer on my next commit, I'm fine with a 15 minute minimum time.

**Comment by VMSolidus** [comment]
image
I'm also adding in configurable variables to let people more easily tweak both the distance it spawns in, and the number of salvage debris it spawns with.

**Comment by VMSolidus** [comment]
- The Listening Outpost now spawns in a debris field of configurable size and density, it is subject to a two-step randomizer so that it is not in the center of said field. 
- The Listening Outpost is still no longer whitelisted, but now it can only spawn after 15 minutes have passed in the round. 
image

**Comment by VMSolidus** [comment]
Alright, awesome. My part of this PR is done, I have finished what I wanted to make for it!

**Comment by VMSolidus** [comment]
@DangerRevolution I just added the mirror to the bathroom. Is there anything else we needed to add, or are we good to get this merged?

**Comment by DangerRevolution** [comment]
Left to do:

- [X] Fix crew monitor not working at mega ranges
- [X] Update the image on this PR to show current listening post.
- [X] Decide hour requirements, current idea is posted in #rebase

**Comment by VMSolidus** [comment]
image
The Listening Outpost has a new appearance. I wanted to make it look a lot comfier for the agents to actually live in long-term.

**Comment by VMSolidus** [comment]
New update Fixes #712

**Comment by VMSolidus** [comment]
image
Alright last picture. I SHOULD be done now.

**Comment by IamVelcroboy** [comment]
Looked over the map changes, made some edits, all is good, 👍

File: Resources/Prototypes/Entities/Objects/Devices/pda.yml
Comment (DebugOk) [inline Resources/Prototypes/Entities/Objects/Devices/pda.yml:1]: First, changing upstream files without a marker. Secondly, not happening. If you want to add this you'll need to make a new item

File: Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/salvage.yml
Comment (DebugOk) [inline Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/salvage.yml]:
```suggestion
  description: "Mindwiped and hypno-indoctrinated into perfect loyalty by the Syndicate. They will throw their life away to defend their objective."
```

**Review by DebugOk — CHANGES_REQUESTED** [review]
I will review the actual code changes when I have time

File: Content.Server/DeltaV/StationEvents/Components/PirateRadioSpawnRuleComponent.cs
Comment (Elijahrane) [inline Content.Server/DeltaV/StationEvents/Components/PirateRadioSpawnRuleComponent.cs]: Remove unused includes

File: Content.Server/DeltaV/StationEvents/Events/PirateRadioSpawnRule.cs
Comment (Elijahrane) [inline Content.Server/DeltaV/StationEvents/Events/PirateRadioSpawnRule.cs]: Then make this a property on the component (the get, set stuff) and clamp it. Unless that's out of fashion now 
image

File: Content.Server/DeltaV/StationEvents/Events/PirateRadioSpawnRule.cs
Comment (Elijahrane) [inline Content.Server/DeltaV/StationEvents/Events/PirateRadioSpawnRule.cs]: TryLoad is a boolean function. Just do (if !_map.TryLoad... to cut out this line

File: Content.Server/DeltaV/StationEvents/Events/PirateRadioSpawnRule.cs
Comment (Elijahrane) [inline Content.Server/DeltaV/StationEvents/Events/PirateRadioSpawnRule.cs]: Loop within a loop isn't *that* uncommon or inappropriate. Remove this comment line

File: Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/salvage.yml
Comment (Elijahrane) [inline Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/salvage.yml]: It's a little weird the modifier set is on them and not their equipment

File: Resources/Prototypes/Entities/Structures/Machines/crew_monitor_server.yml
Comment (Elijahrane) [inline Resources/Prototypes/Entities/Structures/Machines/crew_monitor_server.yml:33]: Am I missing something or will this have side effects for other things? Why was the range limited in the first place, and what consequences for say, other antags, could increasing it have?

File: Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/salvage.yml
Comment (Elijahrane) [inline Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/salvage.yml:103]: Ok I see what's going on. Not sure I like this solution. Are you aware of the 'unremovable' tag? Or whatever drone tools had

Comment (DebugOk) [reply Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/salvage.yml:103]: Even if you give them unremovable guns, they wont actually use them

Comment (VMSolidus) [reply Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/salvage.yml:103]: I cant do this for the gun due to the underlying AI code being unfinished. However I can definitely use the unremoveable tag for their torso. The purpose of which is to make it impossible for players to put the guards in an EVA suit. With no EVA suit, its by extension impossible to kite the guards to the station(where these robust guys will go on a rampage)

File: Resources/Prototypes/Entities/Structures/Machines/crew_monitor_server.yml
Comment (DangerRevolution) [reply Resources/Prototypes/Entities/Structures/Machines/crew_monitor_server.yml:33]: The antag-sided side effect of this would be that; should Nukies get their hands on a crew monitor board, they can see who's dead or alive on the station from their ship and with a mass scanner/radar nearby and a few minutes they could vaguely triangulate their location.

When Listening Post first came out; no matter the range I put it at the syndicate server onboard would pick up the sensors and transmit it to the Crew Monitor aboard the listening post. Doesn't seem to work anymore at our new ranges, hence I've had to crank up the range on the server on the actual station.

File: Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/salvage.yml
Comment (DangerRevolution) [reply Resources/Prototypes/DeltaV/Entities/Mobs/NPCs/salvage.yml:103]: to expand; AI currently do not reload nor find ammo appropriate to their weapon around/in their inventory (all TODO's on the AI code) so this is the best solution atm for it.

**Comment by VMSolidus** [comment]
Awesome, all requested changes by Rane have been made. I enjoy the opportunity to add fun flavor text to places like this. 

image

**Review by VMSolidus — DISMISSED** [review]
While I don't think I should merge this myself without first seeing if Debug gives the okay for this. I've finished my own "review" of this PR, and subsequently performed as much cleanup of it as I could reasonably find. Followed by testing to verify that all features work as intended. I believe that this PR is ready to be merged.

**Review by DebugOk — COMMENTED** [review]
Big scary maths, will require a more in-depth review soon:tm:

**Comment by VMSolidus** [comment]
The big scary maths are about to get even scarier tonight because I have to account for "Stations of Unusual Size". Which I'll do by creating a Constant which is compared against the station size, and then derive a factor that we can use to adjust the final spawning position. Without such a factor, the math breaks down for stations that are significantly larger or smaller than Edge(The reference station)

**Comment by DebugOk** [comment]
Sounds like a lot of work

**Comment by VMSolidus** [comment]
> Sounds like a lot of work

Not too bad really. Since my original reference station for making this block of calculus was Edge, I can literally use Edge as my Constant. All I have to do is load up the spawning script in debug mode, start the map as Edge, spawn the outpost, and record the AABB.Length for Edge. This'll be stored in the spawnrule component as a Constant. To get the Factor for the final position, I can take AABB.Length for the currently spawned station, and divide it by the Constant. That way if the station is either larger or smaller than Edge, it'll move the final position of the outpost by the percentage difference.

**Comment by VMSolidus** [comment]
Since the outpost no longer actually needs its IFF console, I've replaced it with a mass media console.

**Comment by VMSolidus** [comment]
No meaningful changes today, just adding the new license stuff to the PR. Still entirely feature complete, still awaiting final review.

**Review by DebugOk — APPROVED** [review]
I still disagree with some parts but at this point whatever, if tests pass it can be merged

**Pull request merged**
